framework/src/problems/EigenProblem.C

```diff
@@ -347,12 +346,16 @@ EigenProblem::scaleEigenvector(const Real scaling_factor)
   std::vector<VariableName> var_names = getVariableNames();
   for (auto & vn : var_names)
   {
-    MooseVariableFEBase & var = getVariable(0, vn);
-    if (var.eigen())
-      for (unsigned int vc = 0; vc < var.count(); ++vc)
+    MooseVariableBase * var = nullptr;
+    if (hasScalarVariable(vn))
+      var = &getScalarVariable(0, vn);
+    else
+      var = &getVariable(0, vn);
+    if (var->eigen())
+      for (unsigned int vc = 0; vc < var->count(); ++vc)
       {
         std::set<dof_id_type> var_indices;
-        _nl_eigen->system().local_dof_indices(var.number() + vc, var_indices);
+        _nl_eigen->system().local_dof_indices(var->number() + vc, var_indices);
         for (const auto & dof : var_indices)
           _nl_eigen->solution().set(dof, _nl_eigen->solution()(dof) * scaling_factor);
       }
```

Comment:
It looks like this code is duplicated in lines 379-393. Can we make a method for it?

Reply:
Will do

Reply:
Done